internal/deb/extract_test.go

```diff
@@ -352,6 +352,84 @@ var extractTests = []extractTest{{
 		},
 	},
 	error: `cannot extract from package "test-package": path /dir/ requested twice with diverging mode: 0777 != 0000`,
+}, {
+	summary: "Hard link is extracted",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Reg(0644, "./file.txt", "text for file.txt"),
+		testutil.Hln(0644, "./link-to-file.txt", "./file.txt"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/**": []deb.ExtractInfo{{
+				Path: "/**",
+			}},
+		},
+	},
+	result: map[string]string{
+		"/file.txt":         "file 0644 e940b71b",
+		"/link-to-file.txt": "file 0644 e940b71b",
+	},
+	notCreated: []string{},
+}, {
+	summary: "Dangling hard link",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		// The file.txt is left out to create a dangling hard link link-to-file.txt
+		testutil.Hln(0644, "./link-to-file.txt", "./file.txt"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/**": []deb.ExtractInfo{{
+				Path: "/**",
+			}},
+		},
+	},
+	error: `cannot extract from package "test-package": the target file does not exist: .*\/file.txt`,
```

Comment:
Please avoid using `.*`, we want to see the whole error message if it is user-facing (all errors in this PR).